src/doc/trpl/choosing-guarantees.md

```diff
@@ -0,0 +1,417 @@
+% Wrapper Types
+
+// Alternate title: "Choosing your Guarantees"
+
+A recurring theme in Rust programming is that Rust gives one control over the costs and guarantees
+of their program.
+
+There are various "wrapper type" abstractions in the Rust standard library which embody a multitude
+of tradeoffs between cost, ergonomics, and guarantees. Many let one choose between runtime and
+compile time enforcement. This section will thoroughly explain some of these abstractions.
+
+Before proceeding, it is highly recommended that one reads about [ownership][ownership] and
+[borrowing][borrowing] in Rust.
+
+[ownership]: ownership.html
+[borrowing]: references-and-borrowing.html
+
+
+
+# Basic pointer types
+
+## `Box<T>`
+
+[`Box<T>`][box] is an "owned pointer" or a "box". While it can hand out borrowed references to the
+data, it is the only owner of the data. In particular, when something like the following occurs:
+
+```rust
+let x = Box::new(1);
+let y = x;
+// x no longer accessible here
+```
+
+Here, the box was _moved_ into `y`. As `x` no longer owns it, the compiler will no longer allow the
+programmer to use `x` after this. A box can similarly be moved _out_ of a function by returning, and
+when a box (one which hasn't been moved) goes out of scope, destructors are run, deallocating the
+inner data.
+
+This abstraction is a low cost abstraction for dynamic allocation. If you want to allocate some
+memory on the heap and safely pass a pointer to that memory around, this is ideal. Note that you
+will only be allowed to share borrowed references to this by the regular borrowing rules, checked at
+compile time.
+
+[box]: ../std/boxed/struct.Box.html
+
+#### Interlude: `Copy`
+
+Move/ownership semantics are not special to `Box<T>`; it is a feature of all types which are not
+`Copy`.
```

Comment:
covering this again may or may not be worth it. i guess since this is in effective rust it's cool, i just don't like having duplicate docs :)